### PR TITLE
Support running a remote hadoop jar command

### PR DESCRIPTION
**Pull request by interskh**
I am not sure if you guys would like this or whether this could be useful for other people. We have some requirements that needs to submit a hadoop job to a remote server so I added HadoopRemoteJarJobRunner and HadoopRemoteJarJobTask.

### CODE REVIEW

**Comment by erikbern** [comment]
I think this is great but would actually prefer if this is just a setting on the existing Hadoop job class so that everything else works like usual...

**Comment by erikbern** [comment]
just wanted to ping and see what's the status.

is it possible for you to rebase? would also really appreciate it if you can make the changes in the existing `HadoopJobTask` class and just wrap the hadoop streaming command with an `ssh`

**Comment by interskh** [comment]
Yeah I will make the change

**Comment by interskh** [comment]
Hi @erikbern , I've made it in the existing HadoopJarJobTask. How does it look to you now?

**Comment by Tarrasch** [comment]
Looks overall good to me :)

File: luigi/contrib/hadoop_jar.py [hunk 1]
```diff
@@ -64,23 +64,40 @@ def __init__(self):
         pass
 
     def run_job(self, job):
+        ssh_config = job.ssh()
+        if ssh_config:
+            host = ssh_config.get("host", None)
+            key_file = ssh_config.get("key_file", None)
+            username = ssh_config.get("username", None)
```
Comment (Tarrasch) [inline luigi/contrib/hadoop_jar.py]: In the documentation you mentioned these to be required fields. So please don't use `get()` and silently ignore missing fields. Instead use `[]` to make sure they exist.

Comment (interskh) [inline luigi/contrib/hadoop_jar.py]: It is captured later in the if conditions.

Comment (Tarrasch) [inline luigi/contrib/hadoop_jar.py]: Sure, but `[]` would make the logic simpler right? Also you'll see form the stack trace exactly what attribute you're missing.

**Comment by interskh** [comment]
Made the change :)

**Comment by erikbern** [comment]
looks great! i wonder if there's some simple way to test this... probably not

**Comment by Tarrasch** [comment]
Awesome! Anyway, can you find a way to test this @interskh? At least a little bit?

If definitely not, we can merge this I suppose.

**Comment by interskh** [comment]
Looking at it again, I think I might be able to add some basic tests. However, I am not sure how you guys are running tests these days. Is there a doc for that?

**Comment by Tarrasch** [comment]
Ah, finally made this page describing how to run the tests. https://github.com/spotify/luigi/blob/c2ef921ae355e04251dbc0a1673723514a58bcef/CONTRIBUTING.rst

(ping @erikbern, so you don't duplicate the effort)

**Comment by interskh** [comment]
Hi @Tarrasch , I added some basic tests. But the test failed on something that doesn't appear to be related to my change ```ERROR:   pep8: commands failed```

**Comment by erikbern** [comment]
It's the format checker

```
test/contrib/hadoop_jar_test.py:61:5: E303 too many blank lines (2)
```

Great if you can fix that

**Comment by interskh** [comment]
oh lol! how come i didn't catch that.. Let me fix it.

**Comment by interskh** [comment]
updated - let's see if it passes the test this time :sunny:

**Comment by Tarrasch** [comment]
This looks very good to me now! Anyone else want to review?

**Comment by erikbern** [comment]
LGTM let's merge this

**Pull request merged**